lib/Elevate/Database.pm

```diff
@@ -0,0 +1,60 @@
+package Elevate::Database;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Database
+
+Helper/Utility logic for database related tasks.
+
+=cut
+
+use cPstrict;
+
+use Elevate::OS ();
+
+use Cpanel::Pkgr ();
+
+use constant MYSQL_BIN => '/usr/sbin/mysqld';
+
+sub is_database_provided_by_cloudlinux ( $use_cache = 1 ) {
+
+    if ($use_cache) {
+        my $cloudlinux_database_installed = cpev::read_stage_file( 'cloudlinux_database_installed', 0 );
+        return $cloudlinux_database_installed if $cloudlinux_database_installed;
+    }
+
+    return 0 unless Elevate::OS::provides_mysql_governor();
+
+    # Returns undef if database is not provided by cloudlinux
+    my ( $db_type, $db_version ) = Elevate::Database::get_db_info_if_provided_by_cloudlinux();
+
+    return 1 if $db_type && $db_version;
+    return 0;
+}
+
+sub get_db_info_if_provided_by_cloudlinux () {
+    my $pkg = Cpanel::Pkgr::what_provides(MYSQL_BIN);
```

Comment:
```
$>perl -MCpanel::Pkgr -E'say  Cpanel::Pkgr::what_provides("/usr/sbin/mysqld")'             
mysql-community-server
```
I don't follow how you're going to get a db_version out of this?

Reply:
The version is the package name for CL provided versions of mysql/mariadb/percona:

```
# cPperl -MCpanel::Pkgr -E'say  Cpanel::Pkgr::what_provides("/usr/sbin/mysqld")'
cl-MySQL51-server

# cPperl -MCpanel::Pkgr -E'say  Cpanel::Pkgr::what_provides("/usr/sbin/mysqld")'
cl-MariaDB105-server

# cPperl -MCpanel::Pkgr -E'say  Cpanel::Pkgr::what_provides("/usr/sbin/mysqld")'
cl-Percona56-server
```

If it does not match this pattern, then the database server is not provided by CL, and we fallback to cPanel behavior.